R/sequence_coverage.R

```diff
@@ -0,0 +1,35 @@
+#' Sequence coverage
+#'
+#' Calculate sequence coverage for each identified protein
+#'
+#' @param data A dataframe containing at least the protein sequence and the identified peptides as columns.
+#' @param protein_sequence A column containing protein sequences, can be obtained by using the function \code{fetch_uniprot()}
+#' @param peptides A column containing the identified peptides.
+#'
+#' @return A new column containing the calculated sequence coverages for each identified protein
+#' @import dplyr
+#' @importFrom magrittr %>%
+#' @importFrom stringr str_count
+#' @importFrom rlang .data
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' sequence_coverage(
+#' protein_sequence = protein_sequence,
+#' peptides = pep_stripped_sequence
+#' )
+#' }
+#'
+sequence_coverage <-
+  function(data, protein_sequence, peptides)
+{
+    data %>%
+    dplyr::group_by({{protein_sequence}}) %>%
+    find_peptide({{protein_sequence}}, {{peptides}}) %>%
+    dplyr::mutate(sequence_length = nchar({{protein_sequence}})) %>%
+    dplyr::mutate(modified = replace_identified_by_x({{protein_sequence}}, .data$start, .data$end)) %>%
```

Comment:
```suggestion
    dplyr::mutate(modified = replace_identified_by_x({{protein_sequence}}, .data$start, .data$end)) %>%
```
maybe name it modified_sequence. not super necessary but more clear to someone that reads the code and tries to figure out what is going on. Especially since replace_identified_by_x is used.